avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastDeserializerGenerator.java

```diff
@@ -897,46 +902,30 @@ private void processBytes(JBlock body, FieldAction action, BiConsumer<JBlock, JE
   private void processString(Schema schema, JBlock body, FieldAction action,
       BiConsumer<JBlock, JExpression> putValueIntoParent, Supplier<JExpression> reuseSupplier) {
     if (action.getShouldRead()) {
+      JExpression argExprForReadString;
       if (reuseSupplier.get().equals(JExpr._null())) {
-        JExpression readStringInvocation;
-
-        if (schema.getType().equals(Schema.Type.STRING) && !useGenericTypes && SchemaAssistant.isStringable(schema)) {
-          readStringInvocation = JExpr._new(schemaAssistant.classFromSchema(schema))
-              .arg(JExpr.invoke(JExpr.direct(DECODER), "readString").arg(JExpr._null()).
+        argExprForReadString  = JExpr._null();
+      } else {
+        argExprForReadString = JExpr.cast(codeModel.ref(Utf8.class), reuseSupplier.get());
+      }
```

Comment:
I could move the code to the if section.
And I think it is safe to do `Utf8` cast directly since the passed reuse must pass the schema check before invoking this function, which means if the `stringClass` is decided to be `Utf8`, the non-null reuse must be `Utf8` unless there is a bug.
Let me know if I missed anything  important.

Reply:
Hmm... I don't think so. The schema check will ensure that the field is a string, but since the `reuse` object is passed by the application, the field could be populated by any type of `CharSequence`, including either `String` or `Utf8`. Anyway, `instanceof` is quite cheap, from my understanding, and it implicitly includes a null check as well, so I think it'd be better to err on the side of caution and keep the `instanceof`.

We can sync up offline if you want.

Reply:
Make sense, will add back the `instanceof` check.